internal/schemautil/userconfig/stateupgrader/typeupgrader/typeupgrader.go

```diff
@@ -45,18 +29,37 @@ func Slice(s []interface{}, t string) (err error) {
 			continue
 		}
 
-		switch t {
-		case "int":
-			if va == "" {
-				va = "0"
-			}
-
-			s[i], err = strconv.Atoi(va)
-			if err != nil {
-				return err
-			}
+		s[i], err = convert(va, t)
+		if err != nil {
+			return err
 		}
 	}
 
 	return nil
 }
+
+// convert converts a value to the specified type.
+func convert(v string, t string) (res interface{}, err error) {
+	switch t {
+	case "bool":
+		if v == "" {
+			v = "false"
+		}
+
+		res, err = strconv.ParseBool(v)
+		if err != nil {
+			return nil, err
+		}
+	case "int":
+		if v == "" {
+			v = "0"
+		}
+
+		res, err = strconv.Atoi(v)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return res, nil
```

Comment:
The behaviour has changed. Now it fallbacks for an empty interface if `t` is not `bool` or `int`.  
More important that this hadn't been caught by tests. Please provide a test for the case.

Reply:
addressed